From 3853c649047990acb2e85a56c8908a430bf983a3 Mon Sep 17 00:00:00 2001
From: Ytai Ben-Tsvi <ytai@google.com>
Date: Wed, 4 Mar 2020 17:42:53 -0800
Subject: [PATCH 096/297] Relax permission checks in sound trigger middleware

This is a temporary measure to address a bug preventing Now Playing
from working.

Bug: 150750187
Test: Manual
Change-Id: I55f1224ab7301e5cfaff510705d8c7bf21ecfad3
(cherry picked from commit 194969aa5f107440745ada5370df3ff72cc5386e)
---
 .../SoundTriggerMiddlewareValidation.java       | 17 ++---------------
 1 file changed, 2 insertions(+), 15 deletions(-)

diff --git a/services/core/java/com/android/server/soundtrigger_middleware/SoundTriggerMiddlewareValidation.java b/services/core/java/com/android/server/soundtrigger_middleware/SoundTriggerMiddlewareValidation.java
index c45f37dfdbd..49a2901d4f9 100644
--- a/services/core/java/com/android/server/soundtrigger_middleware/SoundTriggerMiddlewareValidation.java
+++ b/services/core/java/com/android/server/soundtrigger_middleware/SoundTriggerMiddlewareValidation.java
@@ -249,21 +249,8 @@ public class SoundTriggerMiddlewareValidation implements ISoundTriggerMiddleware
      * @param permission The permission to check.
      */
     void enforcePermission(String permission) {
-        final int status = PermissionChecker.checkCallingOrSelfPermissionForPreflight(mContext,
-                permission);
-        switch (status) {
-            case PermissionChecker.PERMISSION_GRANTED:
-                return;
-            case PermissionChecker.PERMISSION_HARD_DENIED:
-                throw new SecurityException(
-                        String.format("Caller must have the %s permission.", permission));
-            case PermissionChecker.PERMISSION_SOFT_DENIED:
-                throw new ServiceSpecificException(Status.TEMPORARY_PERMISSION_DENIED,
-                        String.format("Caller must have the %s permission.", permission));
-            default:
-                throw new InternalServerError(
-                        new RuntimeException("Unexpected perimission check result."));
-        }
+        mContext.enforceCallingOrSelfPermission(permission,
+                String.format("Caller must have the %s permission.", permission));
     }
 
     @Override
-- 
2.35.1

